### PR TITLE
rackspace: work around flaky cloud-init/apt update

### PR DESCRIPTION
**Pull request by axw**
## Description of change

When starting instances on RackSpace, cloud-init
frequently fails to resolve the APT archive hosts.
This leads to "apt-get update" failing, which later
leads to (for example) LXD containers not starting.

This appears to be a RackSpace networking issue.
We work around the problem by issuing our own
"apt-get update" as a bootcmd in cloud-init,
followed by another after a sleep if that fails.

## QA steps

1. juju bootstrap rackspace/dfw --config enable-os-upgrade=false --constraints mem=2G
2. juju add-machine -n 10 lxd
(wait for all containers to be provisioned)

Without the change, several containers would reliably fail to come up, and there would be errors in /var/log/cloud-init-output.log for the hosts, saying "Temporary failure resolving 'archive.ubuntu.com'".

## Documentation changes

None.

## Bug reference

Fixes https://bugs.launchpad.net/juju/+bug/1677425

### CODE REVIEW

**Comment by axw** [comment]
[lifecycle-models CI failure](https://trello.com/c/u9ipSvAf/101-lifecycle-models-ci-failure)

File: provider/rackspace/provider_configurator.go [hunk 1]
```diff
@@ -24,13 +26,30 @@ func (c *rackspaceConfigurator) ModifyRunServerOptions(options *nova.RunServerOp
...
 	cloudcfg.AddPackage("iptables-persistent")
+
+	if args.InstanceConfig.EnableOSRefreshUpdate {
+		// cloud-init often fails to update APT caches
+		// during instance startup on RackSpace. Add an
+		// extra call to "apt-get update" with a sleep
+		// on failure to attempt to alleviate this.
+		// See lp:1677425.
+		os, err := jujuseries.GetOSFromSeries(series)
+		if err != nil {
```
Comment (wallyworld) [inline provider/rackspace/provider_configurator.go]: I don't think we should error here - if it's an unknown series, just assume not ubuntu and do nothing.
If the issue truly is rackspace only, here seems as good a place as any. It is a bit kludgy. Not sure off hand of anything better.

Comment (axw) [reply provider/rackspace/provider_configurator.go]: OK, changed (but it will error in other places).

**Comment by axw** [comment]
$$merge$$

**Pull request merged**